src/Nox.Lib/Configuration/NoxUseOptions.cs

```diff
@@ -101,9 +101,12 @@ public void Configure(IApplicationBuilder builder)
         builder.UseWhen(context => context.Request.Path.StartsWithSegments(apiPrefix) && solution.Domain is not null,
             appBuilder =>
             {
-                appBuilder
-                    .UseMiddleware<RelatedEntityRoutingMiddleware>()
-                    .UseMiddleware<ApiRoutingMiddleware>();                    
+                if (solution.Presentation.ApiConfiguration.ApiGenerateRelatedEndpointsMaxDepth > 1 &&
```

Comment:
in the next PR we can add this logic to a static function inside RelatedEntityRoutingMiddleware, `RelatedEntityRoutingMiddleware.IsApplicable(solution)`